### PR TITLE
Filter les enfants de plus de 25 ans non handicapés

### PR DESCRIPTION
**Pull request by fpagnoux**
Connected to #39 
Depends on openfisca/openfisca-france#290

### CODE REVIEW

File: lib/simulation/openfisca/mapping/famille.js [hunk 1]
```diff
@@ -10,9 +11,10 @@ module.exports = {
     },
     enfants: {
         fn: function(situation) {
-            var enfants = _.pluck(_.filter(situation.individus, {role: 'enfant'}), '_id');
-            var personnesACharge = _.pluck(_.filter(situation.individus, {role: 'personneACharge'}), '_id');
-            return enfants.concat(personnesACharge); 
+            var enfants = _.filter(situation.individus, function(individu) {
+                return mapping.isIndividuValid(individu, situation) && individu.role == 'enfant';
+            });
+            return _.pluck(enfants, '_id');
```
Comment (MattiSG) [inline lib/simulation/openfisca/mapping/famille.js]: This code is duplicated all over the place. Now that you've added a dependency to `index.js` for `isIndividuValid`, go all the way and factor in that function  ;)

File: lib/simulation/openfisca/mapping/foyerFiscal.js [hunk 2]
```diff
@@ -11,9 +12,10 @@ module.exports = {
     },
     personnes_a_charge: {
         fn: function(situation) {
-            var enfants = _.pluck(_.filter(situation.individus, { role: 'enfant' }), '_id');
-            var personnesACharge = _.pluck(_.filter(situation.individus, { role: 'personneACharge' }), '_id');
-            return enfants.concat(personnesACharge);
+            var enfants = _.filter(situation.individus, function(individu) {
+                return mapping.isIndividuValid(individu, situation) && individu.role == 'enfant';
```
Comment (MattiSG) [inline lib/simulation/openfisca/mapping/foyerFiscal.js]: I don't get it. This doesn't reads as equivalent to the previous code: we're asking for `personne_a_charge` and returning only children…?

Comment (fpagnoux) [inline lib/simulation/openfisca/mapping/foyerFiscal.js]: In mes-aides today, there is no disctinction between kids and other personnes à charge. To be cleaner, I guess we should remove all references to the role personnesACharge.

File: lib/simulation/openfisca/mapping/index.js [hunk 3]
```diff
@@ -162,8 +162,18 @@ var transform = function(src, schema, situation) {
     return result;
 };
 
+var isIndividuValid = exports.isIndividuValid = function(individu, situation) {
```
Comment (MattiSG) [inline lib/simulation/openfisca/mapping/index.js]: This helper doesn't feel at home in `index.js`. Put it in a separate file.

The best would have been `individu`, but it's already used as a map. What about `common`, maybe?

File: lib/simulation/openfisca/mapping/foyerFiscal.js (hunk 2 above)
Comment (MattiSG) [inline lib/simulation/openfisca/mapping/foyerFiscal.js]: Either remove personnesACharge or make it do what it says. Inconsistency is dangerous.

**Comment by MattiSG** [comment]
The helper function you created would really benefit from a test. I'll give you a headstart into it.

**Comment by MattiSG** [comment]
Here, have a look  :)

**Comment by fpagnoux** [comment]
No ludwig test regression introduced by this PR. Is this good to merge ?

File: lib/simulation/openfisca/mapping/common.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+var _ = require('lodash');
+var moment = require('moment');
+
+var isIndividuValid = exports.isIndividuValid = function(individu, situation) {
```
Comment (MattiSG) [inline lib/simulation/openfisca/mapping/common.js]: Rather declare with
```js
exports.isIndividuValid = isIndividuValid;
function isIndividuValid(…)`
```

You could also group `exports` assignments, either at the end or beginning. But avoid this double assignment, it is uncommon and makes it too easy to miss the export. Plus, it's inconsistent within the file.

File: lib/simulation/openfisca/mapping/common.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    var age = moment(situation.dateDeValeur).diff(moment(individu.dateDeNaissance), 'years');
+    var handicap = _.find(individu.situationsPro, { situation: 'handicap' });
+    return individu.role != 'enfant' || age <= 25 || handicap;
+};
+
+exports.getEnfants= function(situation) {
```
Comment (MattiSG) [inline lib/simulation/openfisca/mapping/common.js]: Always name exported functions (`function getEnfants(…)`). As a general rule, always name functions.

> There are a number of reasons for this, both [pragmatic](http://stackoverflow.com/questions/28353326) and more [convoluted](http://stackoverflow.com/questions/336859). Definite reference [here](https://kangax.github.io/nfe/).

Comment (MattiSG) [inline lib/simulation/openfisca/mapping/common.js]: Missing space before `=`.

File: lib/simulation/openfisca/mapping/foyerFiscal.js [hunk 6]
```diff
@@ -9,12 +10,9 @@ module.exports = {
             }), '_id');
         }
     },
+    // Aujourd'hui, dans mes-aides, tous les enfant et seuls les enfants sont transmis à Openfisca comme personnes à charge
```
Comment (MattiSG) [inline lib/simulation/openfisca/mapping/foyerFiscal.js]: I'd rather have comments, like code, in English. Is it ok with you?

Comment (MattiSG) [inline lib/simulation/openfisca/mapping/foyerFiscal.js]: :+1:

File: lib/simulation/openfisca/mapping/menage.js [hunk 7]
```diff
@@ -1,26 +1,23 @@
 var _ = require('lodash');
+var common = require('./common');
 
 module.exports = {
     personne_de_reference: {
-        fn: function(logement, situation) {
+        fn: function(situation) {
```
Comment (MattiSG) [inline lib/simulation/openfisca/mapping/menage.js]: :+1:

File: lib/simulation/openfisca/mapping/menage.js [hunk 8]
```diff
@@ -30,6 +27,7 @@ module.exports = {
                 'locatairefoyer': 7,
                 'sans_domicile' : 8
             };
+            var logement = situation.logement;
```
Comment (MattiSG) [inline lib/simulation/openfisca/mapping/menage.js]: General note: don't hesitate to navigate objects, adding intermediate variables does not always increase readability.

(no action requested here)

File: test/openfisca/mapping/individu-helpers.coffee [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+      before ->
+          target['situationsPro'] = []
+          target['dateDeNaissance'] = new Date '2010-01'
+
+      it 'should be valid', ->
+        individuHelpers.isIndividuValid(target, SITUATION).should.be.ok
+
+    describe 'over 25', ->
+      before ->
+        target['situationsPro'] = []
```
Comment (MattiSG) [inline test/openfisca/mapping/individu-helpers.coffee]: The order of execution of `describe`, `before` and `it` is the order of declaration, so you can build upon the previous assignments and simply set `dateDeNaissance` here (same below).

File: test/openfisca/mapping/individu-helpers.coffee [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+    it 'should be valid', ->
+      individuHelpers.isIndividuValid(target, SITUATION).should.be.ok
+
+  describe 'a child', ->
+    target =
+      role: 'enfant'
+
+    describe 'under 25', ->
+      before ->
+          target['situationsPro'] = []
```
Comment (MattiSG) [inline test/openfisca/mapping/individu-helpers.coffee]: Why not use the dot notation (`target.situationsPro`)?

**Pull request merged**